src/main/java/net/minestom/server/entity/Player.java

```diff
@@ -947,6 +949,92 @@ public void stopSound() {
         playerConnection.sendPacket(stopSoundPacket);
     }
 
+    /**
+     * Sends the given particle to this player.
+     *
+     * @param particle     the particle to send
+     * @param longDistance whether the client should force this particle to show, even when it is far away
+     * @param x            x position of the particle
+     * @param y            y position of the particle
+     * @param z            z position of the particle
+     * @param offsetX      random x offset of the particle
+     * @param offsetY      random y offset of the particle
+     * @param offsetZ      random z offset of the particle
+     * @param speed        speed of the particle
+     * @param count        amount of times this particle is shown
+     */
+    public void sendParticle(@NotNull Particle particle, boolean longDistance, double x, double y, double z,
+                             float offsetX, float offsetY, float offsetZ, float speed, int count) {
```

Comment:
All this might be nicer to have in a builder. This could be done with providing a vector or x/y/z for the location as well.

Reply:
We could have this in a builder, but it may be better to just have some of those values in the Particle class? Like Particle.setOffset(float, float, float) and the same for count and longDistance.

Reply:
Yeah that's probably a better idea - I'd definitely prefer them being in the class than in a super long method.

Reply:
If the plan is to have a builder, I would prefer `Particle` to be immutable

Reply:
Yeah it isn't right now, but I can change that

Reply:
The problem is that there are several Particle classes (for different kinds of data) so how would the builder construct the right one? There is Particle.block(BlockState) for example, which could return BlockParticleBuilder, but it is not really clean to make a separate builder for each type